packages/client/src/components/gameOverlay/gameOverlay.tsx

```diff
@@ -1,15 +1,27 @@
-import { useState } from 'react'
-
+import React from 'react'
+import { useSelector } from 'react-redux'
 import { GameStart } from '../gameStart/gameStart'
-
+import { GameEnd } from '../gameEnd/gameEnd'
+import { GameStatus } from '../../store/gameSlice'
+import { getGameStatus } from '../../store/selectors'
 import styles from './gameOverlay.module.css'
 
-export const GameOverlay: React.FC = () => {
-  const [isStart, setIsStart] = useState(true)
+interface Props {
+  onReloadGame: () => void
+}
+
+export const GameOverlay: React.FC<Props> = props => {
+  const { onReloadGame } = props
+  const status = useSelector(getGameStatus)
+  const isRender =
+    status === GameStatus.START || status === GameStatus.END ? true : null
```

Comment:
Здесь условие можно упростить
```suggestion
  const isRender = status === GameStatus.START || status === GameStatus.END
```

Reply:
Так это не сработает, TS выкинет ошибку: 
```
Type '(props: Props) => false | Element' is not assignable to type 'FC<Props>'.
  Type 'false | Element' is not assignable to type 'ReactElement<any, any> | null'.
    Type 'boolean' is not assignable to type 'ReactElement<any, any>'.ts(2322)
```

Reply:
Так происходит из-за того, что вы пытаетесь возвратить бул в реакт компоненте. TS реагирует правильно. 

Я бы посоветовал оставить тут рендеринг в любом случае, а поверку на `isRender` и поход в стор вынести в `GameDisplay`. Это будет явно показывать, когда рисуется оверлей (не будет обманчивой видимости из родителя, что он рендерится всегда)

Чтобы получилось в формате 
`{isShowingOverlay ? <GameOverlay /> : null}`